baseplate/file_watcher.py

```diff
@@ -67,11 +91,25 @@ class FileWatcher(object):
         for the watched file to become available (defaults to not blocking).
 
     """
-    def __init__(self, path, parser, timeout=None):
+    def __init__(self, path, parser, timeout=None, **open_options):
+        if sys.version_info.major < 3 and open_options:
+            raise TypeError(
+                "'buffering' is the only supported open_options keyword "
+                "argument for FileWatcher() in Python 2"
+            )
+
+        if "file" in open_options:
+            raise TypeError("'file' is an invalid keyword argument for FileWatcher()")
+        if "mode" in open_options:
+            raise TypeError("'mode' is an invalid keyword argument for FileWatcher()")
```

Comment:
Do we want to support mode?  We could by doing something like:

```
self._open_options.setdefault("mode", "r")
```

and get rid of the `"r"` in the call to open?

Reply:
I don't think we _can_ support `"closefd"` and `"file"` would be confusing to support.

Reply:
Also, if we did support mode, it would have to be restricted to only read types, so we might not want the overhead of that?

Reply:
Maybe a better way if we want to support binary mode is to have `binary=False` and do `"r+b"` if that is set to `True`

Reply:
yeah, agreed. of all the options it looks like `encoding` and `binary` and _maybe_ `newline` are the ones we'd care to let through? probably best to just explicitly add those then? sorry for the bum steer on kwargs.